src/ansys/additive/core/material_tuning.py

```diff
@@ -140,28 +142,37 @@ def __init__(self, input: MaterialTuningInput, msg: MaterialTuningResultMessage,
         os.makedirs(out_dir, exist_ok=True)
 
         self._optimized_parameters_file = os.path.join(out_dir, "optimized_parameters.csv")
-        self._coefficients_file = os.path.join(out_dir, "coefficients.csv")
-        self._material_configuration_file = os.path.join(out_dir, "material_configuration.json")
+        self._coefficients_file = None
+        self._material_configuration_file = None
         self._characteristic_width_file = None
         self._log_file = None
 
         with open(self._optimized_parameters_file, "wb") as f:
             f.write(msg.optimized_parameters)
-        with open(self._coefficients_file, "wb") as f:
-            f.write(msg.coefficients)
-        with open(self._material_configuration_file, "wb") as f:
-            f.write(msg.material_parameters)
 
-        if len(msg.characteristic_width_lookup) > 0:
+        if msg.coefficients and len(msg.coefficients) > 0:
```

Comment:
Checking for a list with just `if msg.coefficients:` implies len > 0.
```suggestion
        if msg.coefficients:
```

Reply:
Updated all if statements containing lists.